Add the ability to run upgrade test

red1452 opened the pull request:
Upgrade test runs after regression test. It requires to use tar-archive with previous versions of diskquota, which was generated from last release. This archive is generated at `diskquota_artifacts` and named `diskquota-<VERSION>-rhel7_x86_64.tar.gz` Path to this archive is set in command to run Docker: `<PATH_TO_ARCHIVE_WITH_OLD_VERSIONS>`.  At this archive all `*.so` files must be at `lib/postgresql` path:
```
evgeniy@evgeniy-pc:~/gpdb/diskquota_bin$ tar -tvf diskquota-2.2.1-rhel7_x86_64.tar.gz 
-rwxr-xr-x gpadmin/gpadmin  46 2023-10-03 13:59 install_gpdb_component
-rw-r--r-- gpadmin/gpadmin 202 2023-10-05 17:02 diskquota-build-info
drwxrwxr-x gpadmin/gpadmin   0 2023-10-05 17:02 lib/
drwxrwxr-x gpadmin/gpadmin   0 2023-10-05 17:02 lib/postgresql/
-rwxr-xr-x gpadmin/gpadmin 281632 2023-10-03 13:55 lib/postgresql/diskquota.so
-rwxr-xr-x gpadmin/gpadmin 737008 2023-10-05 16:49 lib/postgresql/diskquota-2.1.so
-rwxr-xr-x gpadmin/gpadmin 550080 2023-10-03 13:53 lib/postgresql/diskquota-2.0.so
-rwxr-xr-x gpadmin/gpadmin 755664 2023-10-05 17:02 lib/postgresql/diskquota-2.2.so
```

Before run tests it is needed to run build step:
```
docker run --rm -it -e DISKQUOTA_OS=rhel7 \
       -v /tmp/diskquota_artifacts:/home/gpadmin/diskquota_artifacts \
       -v <PATH_TO_DISKQUOTA_SRC>:/home/gpadmin/diskquota_src \
       -v <PATH_TO_CMAKE_INSTALL_SCRIPT>:/home/gpadmin/bin_cmake/cmake-3.20.0-linux-x86_64.sh \
       -v <PATH_TO_ARCHIVE_WITH_OLD_VERSIONS>:/home/gpadmin/last_released_diskquota_bin/diskquota-<VERSION>-rhel7_x86_64.tar.gz \
       hub.adsw.io/library/gpdb6_regress:latest diskquota_src/concourse/scripts/entry.sh build
``` 

Command to run all tests in Docker is not changed:
```
docker run --rm -it --sysctl 'kernel.sem=500 1024000 200 4096' \
       -v /tmp/diskquota_artifacts:/home/gpadmin/bin_diskquota \
       -v <PATH_TO_DISKQUOTA_SRC>:/home/gpadmin/diskquota_src \
       -v <PATH_TO_CMAKE_INSTALL_SCRIPT>:/home/gpadmin/bin_cmake/cmake-3.20.0-linux-x86_64.sh \
       hub.adsw.io/library/gpdb6_regress:latest diskquota_src/concourse/scripts/entry.sh test
```

After releasing new version of diskquota it should to get new archive from diskquota_artifacts and use it at tests for next pull requests.

RekGRpth commented on src/diskquota.c:
```diff
@@ -381,7 +381,7 @@ disk_quota_worker_main(Datum main_arg)
 		int has_error = worker_spi_get_extension_version(&major, &minor) != 0;
 
-		if (major == DISKQUOTA_MAJOR_VERSION && minor == DISKQUOTA_MINOR_VERSION) break;
+		if (major == DISKQUOTA_MAJOR_VERSION) break;
```
I think we shouldn't do that!

RekGRpth replied:
Suppose the user had version 2.1 installed and upgraded the library to version 2.2.
Then, at startup, diskquota will issue a warning and offer to update the extension.
But with this patch, the diskquota will continue to work with the old SQL part.

red1452 replied:
If user will use our version of diskquota, diskqouta version 2.1 will be the same as diskquota version 2.2, because diskquota 2.1 is symlink of diskquota 2.2.

RekGRpth replied:
Yes, but in the database the user has version 2.1 before the update and he does not know that he needs to update!

red1452 committed:
Revert "temporary disable upgrade tests"

This reverts commit 50ed2e4e1883ec8ec4e7086b750cb28cdc5a2dc0.

red1452 committed:
Add check to creating symlinks (#21)

If previous versions of diskquota were added from DISKQUOTA_LAST_RELEASE_PATH
do not need to create symlinks from ealier versions (since 2.0)
diskquota to the latest version which added at commit 472eb0652d7d71d8605a70b9f8f15005009ae1c9

New check disable creating symlinks if diskquota was build with -DDISKQUOTA_LAST_RELEASE_PATH
and enable creating symlinks if there is not -DDISKQUOTA_LAST_RELEASE_PATH

red1452 committed:
Update README.md for upgrade tests (#21)

Information about how to run upgrade tests for diskquota was added
to arenadata/README.md.

red1452 committed:
Fix check for find changes at DDL files (#21)

After commit 97f1f9b46b0941a86cffdb2120b900b55ce58928 all DDL files
were moved to directory "control/ddl". At upgrade test there is check,
which finds diff at DDL files from last release version. Path, where
DDL files are found, was not changed, and this check did not work.

red1452 committed:
fix get the latest tag of current branch

red1452 committed:
remove waste symbol
